docs/extend/packaging-zos-extensions.md

```diff
@@ -0,0 +1,95 @@
+    |-- configure.sh
+    |-- jobs-api-server-1.0.4-boot.jar
+    |-- start.sh
+    |-- validate.sh
+```
+
+- `manifest.yaml`: Zowe component manifest file. Detail definition of manifest can be found in [Zowe Component Manifest](#zowe-component-manifest) section.
+- `apiml-static-registration.yaml.template`: Supporting file to instruct Zowe launch script how to register this extension service to API Mediation Layer Discovery service. In this case, this file is referred in `manifest.yaml` `apimlServices.static[0].file` field. This file is optional based on the function of the component and the file name can also be changed and customized in manifest file.
+- `bin/(configure|start|validate).sh`: These are Zowe component lifecycle scripts. You may not need these files based on the function of the component. Detail definition of lifecycle scripts can be found in [Zowe lifecycle](#zowe-runtime-lifecycle) section.
+
+We also suggest to put other files into the package:
+
+- `README.md`: a brief introduction of your extension in Markdown format, like how it should be installed, configured, verified, etc.
+- `LICENSE`: full license text file.
+
+If you decide to bundle and ship Zowe extension(s) within another product. You can put the whole directory structure above into your product package as sub-directories. For example:
+
+```
++-- <my-product-root>
+    |-- <other-directories-and-files>
+    |-- zowe-extension-A
+        |-- manifest.yaml
+        |-- bin/
+            |-- start.sh
+    |-- zowe-extension-B
+        |-- manifest.yaml
+        |-- bin/
+            |-- start.sh
+```
+
+## Zowe component manifest
+
+_Note: this feature is added with Zowe v1.18.0 release._
+
+Zowe extensions, and as well as core components, can use a manifest file to describe itself. The manifest file defines the name, purpose of the component, and it also provides information how this component should be installed, configured, started, and tested. It can be named as `manifest.yaml`, `manifest.yml` or `manifest.json` and should locate in the root directory of this component. Currently only `YAML` or `JSON` format are supported.
+
+The properties supported by the manifest file are listed here:
+
+- **`name`**: This is required. It defines a short, computer readable name of the component. This component name will be used as directory name your component after it's installed. The allowed characters in the name are alphabets, numbers, hyphen (`-`) and underscore (`_`). For example, `explorer-jes` is a valid extension name.
+- **`id`**: This is optional. It defines a long, computer readable identifier of the component. The identifier also matches the component path in Zowe Artifactory. For example, `org.zowe.explorer-jes` is a valid identifier, we can locate the component official releases by looking into `libs-release-local/org/zowe/explorer-jes/` directory of [Zowe Artifactory](https://zowe.jfrog.io/ui/repos/tree/General/libs-release-local%2Forg%2Fzowe%2Fexplorer-jes).
```

Comment:
It might be worth mentioning that the artificatory info is only relevant to OMP hosted projects?